stix2/utils.py

```diff
@@ -25,8 +26,10 @@
 # STIX object properties that cannot be modified
 STIX_UNMOD_PROPERTIES = ['created', 'created_by_ref', 'id', 'type']
 
-TYPE_REGEX = r'^\-?[a-z0-9]+(-[a-z0-9]+)*\-?$'
-SCO21_EXT_REGEX = r'^\-?[a-z0-9]+(-[a-z0-9]+)*\-ext$'
+TYPE_REGEX = re.compile(r'^\-?[a-z0-9]+(-[a-z0-9]+)*\-?$')
+TYPE_21_REGEX = re.compile(r'^\-?([a-z][a-z0-9]*)+(-[a-z0-9]+)*\-?$')
+EXT_21_REGEX = re.compile(r'^\-?([a-z][a-z0-9]*)+(-[a-z0-9]+)*\-ext$')
+PREFIX_21_REGEX = re.compile(r'^[a-z].*')
```

Comment:
If the 2.1 rule is that all type names must start with a letter, then `TYPE_21_REGEX` shouldn't start with `^\-?`.  That allows matching strings to start with a hyphen.  Maybe `EXT_21_REGEX` too, if extension names count as "types".

Reply:
I didn't add the `\-?`.  It was part of the re before I made any changes.  I guess we should determine if it should be there.  This is related to a comment above...

Reply:
`\-?` shoudn't be there for 2.1.